aquadoggo/src/network/service.rs

```diff
@@ -270,6 +277,40 @@ pub async fn network_service(
                         }
                     }
                 }
+                SwarmEvent::Behaviour(BehaviourEvent::RelayServer(event)) => {
+                    debug!("Unhandled relay server event: {event:?}")
+                }
+                SwarmEvent::Behaviour(BehaviourEvent::RelayClient(event)) => {
+                    debug!("Unhandled relay client event: {event:?}")
+                }
+                SwarmEvent::Behaviour(BehaviourEvent::Autonat(event)) => {
+                    match event {
+                        autonat::Event::StatusChanged { old, new } => {
+                            trace!("NAT status changed from {:?} to {:?}", old, new);
+
+                            if let Some(addr) = external_circuit_addr.clone() {
```

Comment:
Previously I was only initiating this set of actions (add external circuit address and register with the rendezvous server) if the NAT status changed to Private. However, since public NAT status does not necessarily mean that direct connectivity is possible, I am initializing the relay client behaviour for any NAT status if `--relay-client` mode is enabled.

Reply:
Does a nat status change mean our reachable address has changed?

Reply:
> Does a nat status change mean our reachable address has changed?

Not quite. A NAT status change occurs when we receive data from a peer which tells us that our presumed public address is either reachable or not reachable. So the starting state is `Unknown`; this shifts to `Public` or `Private` based on insight passed to us by a peer (as part of the `autonat` protocol).

I was originally only executing the relay client-related code if / when the status changed to `Private` (ie. "oh, we're not publically accessible, we better register our circuit-relayed address so other peers can reach us via relayed connections"). However, I noticed that a `Public` NAT status does not always mean that a direct connection is possible (this is the case on my laptop, for example). So then I decided to simply execute the relay client-related code after the first status change, as long as relay client mode is active.

So, technically, the registration code-block could be moved elsewhere.

Reply:
Thanks for clarifying!